web/js/codeworld.js

```diff
@@ -843,14 +843,17 @@ function newFolder() {
 }
 
 function formatSource() {
-  if (window.buildMode === 'codeworld') {
+  const { codeworldEditor, buildMode } = window;
+  const { doc } = codeworldEditor;
```

Comment:
Please use `getDoc()` here.  The `doc` property isn't documented, so it's not part of the public API.